### ISSUE TITLE
CI Status Indicator Shows False Positives

### ISSUE THREAD

**Issue by nicholas-codecov** [codecov/engineering-team#1406]
### Description

Currently our `CIStatus` component defaults to showing a "Processing" state in the even if there is no CI being used. This is an issue because it seems like something is still happening in the background when there really isn't. Here's an example of the commit page when using bundle analysis locally:

![Screenshot 2024-03-13 at 07 21 41](https://github.com/codecov/engineering-team/assets/105234307/1bad299c-3722-4f5c-a2a1-363e21b55e1f)

You can view an example [here](https://app.codecov.io/gh/nicholas-codecov/demo-solid-app/commit/d85d0962247e1916fb9a2a60fff3323d2d25fac7)

## Possible solution

We currently send the service name alongside some other information to the API (at least in the bundle analysis plugins). If this service name is `Local` we can safely assume that the user probably isn't user CI for this upload, and therefore we should remove this indicator altogether. We should update the current CI status field in the API to not just be a boolean value because that does not represent the possible states. We should instead move to a GraphQL Enum that represents the different possible states, that way we can handle the case where the user has uploaded something locally where there is no CI state that will ever be set, and we can just hide this element in the UI.

![Screenshot 2024-03-28 at 9 44 00 AM](https://github.com/codecov/engineering-team/assets/152432831/e0d92e8a-5aec-4842-a22c-6c2e8f9e489c)

Figma: [link](https://www.figma.com/file/szf5HaMolqn6TeAFRZSawS/GH-1406?type=design&node-id=53-772&mode=design&t=5Ri2AQri9iKCf5mO-4)
```[tasklist]
### Related issue
```

**Comment by codecovdesign** [comment]
related: https://github.com/codecov/internal-issues/issues/121

**Comment by Adal3n3** [comment]
Adding figma [file](https://www.figma.com/file/szf5HaMolqn6TeAFRZSawS/GH-1406?type=design&node-id=0-1&mode=design) here for review. The review 🧵is on [slack](https://sentry.slack.com/archives/C04MTDH3ZQA/p1711060579537739?thread_ts=1710974354.622249&cid=C04MTDH3ZQA).

**Comment by Adal3n3** [comment]
@nicholas-codecov @katia-sentry  

Design updated 3/22: (Figma file is same as above)

1. Pending emoji updated.
2. Added arial labels: 
    ✅ aria-label="complete"
   ⏳ aria-label="pending"
   ❌ aria-label="error"
3. Added CI status icons on the dropdown selection for consistency.

![Screenshot 2024-03-22 at 11 46 10 AM](https://github.com/codecov/engineering-team/assets/152432831/f0c745df-55ac-462e-ba4f-009d5ca2c9d1)

**Comment by nicholas-codecov** [comment]
My understanding is that we do not currently record the CI Status, we only track internal processing status, so we will need to work with the platform team on that. Also adding in these selects for the updated CI status, and bundles will need API work as well.

**Comment by Adal3n3** [comment]
Thanks @nicholas-codecov for the note. Sounds like only the "Coverage upload status" filter can move forward the rest need to wait. 

@giovanni-guidini how can we record the CI status?

**Comment by nicholas-codecov** [comment]
Just wondering what the actual expectation is for this ticket atm, are we adding in another column to the pulls and commits table?

If there hasn't been a coverage upload, we already have messaging to indicate that in the coverage, patch, and change columns

**Comment by codecovdesign** [comment]
- here is a related issue that is fundamental to the problem: https://github.com/codecov/internal-issues/issues/121, the initial step could be updating the copy ("upload status" vs "CI status"). This one issue has been slipping; so could something like this be our intro here?
- if that works for our initial iteration, I suggest we bring this back to `in discovery`and breakdown the pieces a bit (column, bundle, CI status checking etc).

@katia-sentry @nicholas-codecov @Adal3n3 wdyt? 🤔 

<img width="838" alt="Screenshot 2024-03-27 at 8 50 15 AM" src="https://github.com/codecov/engineering-team/assets/87861036/91673467-728b-4e47-8789-8c19a5ec94bc">

**Comment by nicholas-codecov** [comment]
We also need to update the "CI Passed" check on both the commits and pulls pages respectively

**Comment by codecovdesign** [comment]
@nicholas-codecov noted in the other issue - thanks.  just to confirm, these spots, right?

<img width="988" alt="Screenshot 2024-03-27 at 9 01 56 AM" src="https://github.com/codecov/engineering-team/assets/87861036/346d02af-dc8d-4f75-aa92-bd869da130b4">
<img width="801" alt="Screenshot 2024-03-27 at 9 12 05 AM" src="https://github.com/codecov/engineering-team/assets/87861036/35cc31af-76b3-4931-97c3-04fb5217e91c">

**Comment by nicholas-codecov** [comment]
I believe that's it yes

**Comment by Adal3n3** [comment]
@codecovdesign @nicholas-codecov 

> the initial step could be updating the copy ("upload status" vs "CI status"). This one issue has been slipping; so could something like this be our intro here?

Based on the definition of each state below: 
- Complete: Upload completed
- Pending: we have received the report, but haven’t started processing it.
- Error: Error would be something wrong with the report, it’s empty, it’s got an unsupported format
- Skipped: the CI skipped running for that commit. No tests run, no reports where uploaded (probably)

We can update the copy to "Coverage upload status", and show "complete", "pending", "error", remove the "skipped" bc it doesn't belong to the upload status.

![Screenshot 2024-03-27 at 11 26 34 AM](https://github.com/codecov/engineering-team/assets/152432831/ace30c9e-db94-4439-a7bf-089bb25d550a)

and then drop the CI status filter bc it's misleading, we don't record the CI status.

**Comment by codecovdesign** [comment]
@Adal3n3 makes sense - looks good!
What issue should be used for the above iteration? We could use https://github.com/codecov/internal-issues/issues/121 (or close), based on [this update](https://github.com/codecov/engineering-team/issues/1406#issuecomment-2023775903) or create sub-issue here to resolve; just looks like there are other areas still under exploration.

**Comment by Adal3n3** [comment]
@codecovdesign Thanks for calling this out. I should have used https://github.com/codecov/internal-issues/issues/121 for the "Coverage upload status" filter and use this issue for updating the copy and status icon from "processing" to "no status"(on below screenshot) 

![Screenshot 2024-03-27 at 2 44 54 PM](https://github.com/codecov/engineering-team/assets/152432831/26c79f21-c48e-40ef-aaef-06809952e719)

**Comment by codecovdesign** [comment]
@Adal3n3 so 121 confirms the dropdown copy changes and this one will do the status seen on individual commits / pr + PR > pulls? The other exploration I was mentioning was the question about CI checks etc - maybe that is a separate issue? Any way sounds good just wanting to capture the `dev ready` changes for handoff vs later exploration items.

**Comment by katia-sentry** [comment]
@codecovdesign @Adal3n3 @nicholas-codecov Prioritized https://github.com/codecov/internal-issues/issues/121 for upcoming sprint. We can bring this issue back to `in-discovery`.

**Comment by katia-sentry** [comment]
@codecovdesign @Adal3n3  how is this issue different from 121?  cc. @nicholas-codecov

**Comment by Adal3n3** [comment]
@katia-sentry @codecovdesign Sorry all for the confusion. To clarify, the difference between this issue and issue 121 is the location of the copy change/fix.

This issue is for updating the copy and status icon from "processing" to "no status" or like @nicholas-codecov suggested above on the description, hiding the UI element when where there is no CI state.

![Screenshot 2024-03-28 at 9 44 00 AM](https://github.com/codecov/engineering-team/assets/152432831/28865c00-cf21-41b4-af6f-9482aafb67e0)

I bring this issue back in-discovery, and will tag you all to review the change on slack.

I think we should create separate issues for CI status checking, bundle upload status, etc as they require some API works.

**Comment by Adal3n3** [comment]
No action item. Just want to add another example to this issue.

As long as we remove/hide the "processing" status, or change it to "No status" we will prevent further confusion.

https://sentry.zendesk.com/agent/tickets/117502
![Screenshot 2024-04-09 at 10 49 51 AM](https://github.com/codecov/engineering-team/assets/152432831/7b516cdc-6616-41be-98c7-587d76bab921)

**Comment by nicholas-codecov** [comment]
Looking a bit deeper into the API, it seems like our current "CI Passed" value is only a `boolean` or `undefined / null`, we do not actually have a "state" to use.

So currently it looks like this:

- `null | undefined` -> Pending
- `false` -> Failed
- `true` -> Processed

I think we need to revisit the API and adjust this accordingly before we can take any action, unless we're just getting rid of the processing state altogether.

**Comment by nicholas-codecov** [comment]
Ignore my previous comment, I'm just now seeing your ticket for the API changes.
